actions/translation-cursor-deploy/main.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Custom Cursor Deploy GitHub action
+ * @see {@link https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action}
+ *
+ * Updates the deployment translation cursor file in the S3 bucket.
+ */
+
+import * as core from '@actions/core'
+import {exec} from '@actions/exec'
+
+import {writeLineToFile, runAction, fileExistsInS3} from '../utils'
+
+runAction(async () => {
+    const bucket = core.getInput('bucket_name', {required: true})
+    const hash = core.getInput('hash', {required: true})
+
+    await cursorDeploy({
+        bucket,
+        hash
+    })
+})
+
+const PREVIOUS_HASH_COMMAND = 'previous' // special command for rollback to hash from /previous
+
+interface CursorDeployActionArgs {
+    bucket: string
+    hash: string
+}
+
+export async function cursorDeploy({bucket, hash}: CursorDeployActionArgs) {
+    const previousPath = `s3://${bucket}/translation-deploy/previous`
+    const latestPath = `s3://${bucket}/translation-deploy/latest`
+
+    if (hash === PREVIOUS_HASH_COMMAND) {
+        const isPreviosFileExists = await fileExistsInS3({
+            bucket,
+            key: 'translation-deploy/previous'
+        })
+        if (!isPreviosFileExists) {
+            core.setFailed('Previous cursor is empty')
+            return
+        }
+
+        const isLatestFileExists = await fileExistsInS3({bucket, key: 'translation-deploy/latest'})
+        if (isLatestFileExists) {
+            await exec('aws s3 rm ', [latestPath])
+        }
+        await exec('aws s3 cp ', [previousPath, latestPath])
```

Comment:
I suggest using the helper method for copying to S3 https://github.com/pleo-oss/pleo-spa-cicd/blob/main/actions/utils.ts#L64 - it will make it easier to mock during testing.